### PR TITLE
test: migrate existing tests to jest

### LINKED ISSUE #107: Migrate to Jest
To migrate existing tests, run the following command: 
```zsh
$ npx jest-codemods
```
See details in  https://github.com/skovhus/jest-codemods

To install jest, just issue the following command:
```zsh
$ npm i -D ts-jest jest @types/jest
```
Once it is done, generate the config:
```zsh
$ npx ts-jest config:init
```

### PR DESCRIPTION
**Pull request by denis-maiorov-brightsec**
closes #107

### CODE REVIEW

**Review by derevnjuk — CHANGES_REQUESTED** [review]
1. Use `toMatchObject` where it is possible

Before:
```ts
// ...
expect(har).toHaveProperty('log.entries[0].response.status', 200);
expect(har).toHaveProperty('log.entries[0].response.content.size', 4);
// ...
```
After:
```ts
expect(har).toMatchObject({
  log: {
    entries: [
      {
        request: {
          url,
          method: 'GET',
          headers: [{ name: 'host', value: 'localhost:8000' }]
        },
        response: {
          status: 200,
          content: { size: 4, mimeType: 'text/plain', text: 'body' }
        }
      }
    ]
  }
});
```
2. Use `expect(result).toBeUndefined()` instead of `expect(typeof result).toEqual('undefined')`
3. You have to test specific expection where it is neccessary
Original:

```ts
(() => parseUrl(input)).should.throw('Invalid URL');
```

Incorrect:

```ts
expect(() => parseUrl(input)).toThrowError(Error);
```

Correct:

```ts
expect(() => parseUrl(input)).toThrowError('Invalid URL');
```

In case when you expect some instance of the derived class, just put a couple of assertions:

```ts
expect(() => parseUrl(input)).toThrowError('Invalid URL');
expect(() => parseUrl(input)).toThrowError(TypeError);
```

File: packages/editor/tests/oas2.spec.ts [hunk 1]
```diff
@@ -258,8 +257,8 @@ describe('OasV2Editor', () => {
       it('should serialize yaml into yaml', () => {
         const result = openApiEditor.stringify();
 
-        result.should.be.a('string');
-        result.should.match(/^swagger:/);
+        expect(typeof result).toBe('string');
```
[the remark this answers is not in the document]
Comment (denis-maiorov-brightsec) [reply packages/editor/tests/oas2.spec.ts:260]: It actually doesn't work:
```
 expect(received).toBeInstanceOf(expected)

    Expected constructor: String

    Received value has no prototype
    Received value: "lorem"
```

I think we should stick to typeof check bc imo using String constructor to construct string complicates things and should never be used

[the remark this answers is not in the document]
Comment (derevnjuk) [reply packages/editor/tests/oas2.spec.ts:260]: Yeah, you are right. It won't work with primitives

File: packages/validator/tsconfig.spec.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "./tsconfig.json"
+}
```
Comment (derevnjuk) [inline packages/validator/tsconfig.spec.json:3]: You don't need to create a dedicated `tsconfig.spec.json`,  `tsconfig.json` already includes tests by default

File: packages/validator/tests/oas.spec.ts [hunk 3]
```diff
@@ -41,15 +41,15 @@ describe('OASValidator', () => {
...
+      expect(result).toMatchObject([]);
     });
 
     it('should successfully validate valid oas v3 document (spoonacular, json)', async () => {
       const input: OpenAPIV3.Document = spoonacularOas as OpenAPIV3.Document;
 
       const result = await validator.verify(input);
 
-      expect(Object.keys(result)).toHaveLength(0);
+      expect(result).toMatchObject([]);
```
Comment (derevnjuk) [inline packages/validator/tests/oas.spec.ts:52]: Please fix the same issue in other places as well

```suggestion
      expect(result).toEqual([]);
```

**Review by derevnjuk — CHANGES_REQUESTED** [review]
BTW Let's create a root `jest.config.js` as follows:

```js
module.exports = {
  "projects": ["<rootDir>/packages/*"]
};
```

File: .eslintignore [hunk 4]
```diff
@@ -4,3 +4,5 @@
 tmp
 *-lock.json
 *.lock
+jest.*.js
+**/jest.config.js
```
Comment (derevnjuk) [inline .eslintignore:8]: New line at end of the file is expected)

File: package.json [hunk 5]
```diff
@@ -44,7 +43,7 @@
     "test": "lerna run test",
     "test:changed": "lerna run --since origin/master --include-dependents test",
     "coverage": "lerna run coverage",
-    "coverage:all": "nyc npm t",
+    "coverage:all": "lerna run coverage",
```
Comment (derevnjuk) [inline package.json:46]:
```suggestion
    "coverage:all": "jest --converage",
```

**Review by derevnjuk — COMMENTED** [review]
Please keep in mind as soon as we changed the reports folder, CI wouldn't upload them to CC anymore.

You have to update path to reports here:

https://github.com/NeuraLegion/har-sdk/blob/f5be7ae58a5720be8f6a0a8bff7df6556f428f41/.github/workflows/auto-build.yml#L69-L73

File: packages/core/tests/YamlLoader.spec.ts [hunk 6]
```diff
@@ -12,16 +12,16 @@ describe('YamlLoader', () => {
...
     const expectedExceptionMessage = `bad indentation of a mapping entry (5:2)\n\n 2 |   bar: 42`;
 
     // assert
-    (() => loader.load(input)).should.throw(
-      YAMLException,
-      expectedExceptionMessage
-    );
-    expectedExceptionMessage.should.include(
-      loader.getSyntaxErrorDetails().message
+    expect(() => loader.load(input)).toThrowError(YAMLException);
```
Comment (derevnjuk) [inline packages/core/tests/YamlLoader.spec.ts:22]:
```suggestion
    expect(() => loader.load(input)).toThrowError(YAMLException);
    expect(() => loader.load(input)).toThrowError(expectedExceptionMessage);
```

**Review by derevnjuk — CHANGES_REQUESTED** [review]
You have to revert package-lock.json, install npm@6 and update install deps again. The expected version is v1, we are not going to migrate to v2 within this PR)

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**